s3transfer/utils.py

```diff
@@ -179,8 +179,60 @@ def rename_file(self, current_filename, new_filename):
         rename_file(current_filename, new_filename)
 
 
+class DeferredOpenFile(object):
+    OPEN_METHOD = open
```

Comment:
Flagging for opinions on use of class variable that allows you to control how the file is opened. I can see being able to switch how you open the file as being useful, but is not getting exercised. Really only switch it out in the tests. Without a way of injecting an open method there is not a great way to test it other than patching, which I think is the least desirable option.